refactor: more efficient findHook

tkurki opened the pull request:
Require the hooks once upon startup, foregoing the synchronous
file system access for each sentence.

joabakk commented:
Simple and elegant! I started on a more complicated solution, but was interrupted. I guess this should also fetch the sub-hooks under ALK (for STALK messages) as mentioned in #102 ?

tkurki commented:
yes, go ahead and fix that as well, please

joabakk commented:
Not sure it helps a lot on speed. I ran `time cat nmea.4.log | signalk-parser-nmea0183/bin/nmea0183-signalk > output.log` and got `> output.log  0.99s user 0.94s system 27% cpu 6.956 total` for the master branch and `> output.log  1.05s user 0.90s system 28% cpu 6.863 total` for  this PR. Is this a valid speed test?

tkurki commented:
Fixed the bug that was causing `bin/nmea0183-signalk` to not output anything and tested:
```
$ time bin/nmea0183-signalk <../server-node/samples/plaka.log  >/dev/null

real	0m2.680s
user	0m2.526s
sys	0m0.290s

$ git co improve-findhook 
Switched to branch 'improve-findhook'
$ time bin/nmea0183-signalk <../server-node/samples/plaka.log  >/dev/null

real	0m1.518s
user	0m1.552s
sys	0m0.081s
```
